src/audio_decoder_midi.cpp

```diff
@@ -0,0 +1,405 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <array>
+#include <algorithm>
+#include "audio_decoder_midi.h"
+#include "midisequencer.h"
+#include "output.h"
+
+using namespace std::chrono_literals;
+
+constexpr int AudioDecoderMidi::midi_default_tempo;
+
+// 1 ms of MIDI message resolution for a 44100 Hz samplerate
+constexpr int samples_per_play = 512;
```

Comment:
If the sample rate is guaranteed  to be the same for current settings, it could be based on sample rate setting somehow to prevent precision loss when playing at 22050 Hz or less.

Reply:
Fixed this now with a branch for the common values (22050 and 11025). For 44100 vs 48000 the difference doesn't really matter.